app/main.py

```diff
@@ -1,2 +1,77 @@
+from __future__ import annotations
+from typing import Any
+from dataclasses import dataclass
+
+
+@dataclass()
+class Node:
+    key: Any
+    key_hash: int
+    value: Any
+
+
+class ResizeManager:
+    def __init__(
+            self,
+            dictionary: Dictionary
```

Comment:
The `Dictionary` class is used here before it is defined. This will cause a `NameError`. Consider defining the `Dictionary` class before using it in the `ResizeManager` class.